driver-core/src/main/com/mongodb/internal/time/StartTime.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.internal.time;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @see TimePoint
+ */
+public interface StartTime {
+
+    /**
+     * @see TimePoint#elapsed()
+     */
+    Duration elapsed();
+
+    /**
+     * @see TimePoint#asDeadline()
+     */
+    Deadline asDeadline();
+
+    /**
+     * Returns an {@linkplain Deadline#isInfinite() infinite} deadline if
+     * {@code timeoutValue} is negative, an expired deadline if
+     * {@code timeoutValue} is 0, otherwise a deadline in {@code durationNanos}.
+     * <p>
+     * Note that some code might ignore a deadline, and attempt to perform
+     * the operation in question at least once.</p>
+     * <p>
+     * Note that the contract of this method is also used in some places to
+     * specify the behavior of methods that accept {@code (long timeout, TimeUnit unit)},
+     * e.g., {@link com.mongodb.internal.connection.ConcurrentPool#get(long, TimeUnit)},
+     * so it cannot be changed without updating those methods.</p>
+     *
+     * @see TimePoint#fromNowOrInfiniteIfNegative(long, TimeUnit)
+     */
+    Deadline fromNowOrInfiniteIfNegative(long timeoutValue, TimeUnit timeUnit);
```

Comment:
I think perhaps this method should not be added, but I have included it because otherwise, it seems I would need to change existing locations that take long/TimeUnit in this PR. I have at least made the name long and noticeable.